docs/UserGuide.md

```diff
@@ -41,146 +31,257 @@ By : `Team SE-EDU`  &nbsp;&nbsp;&nbsp;&nbsp; Since: `Jun 2016`  &nbsp;&nbsp;&nbs
+
+> Undo the last data-mutating command inputted by the user.<br>
+> If a number is entered, undoes that ammount of previous commands instead.<br>
+> The number **must be a positive integer** 1, 2, 3, ...
 
 Examples:
 
-* `list`<br>
-  `select 2`<br>
-  Selects the 2nd person in the address book.
-* `find Betsy` <br>
-  `select 1`<br>
-  Selects the 1st person in the results of the `find` command.
+* `add Test`<br>
+  `undo`<br>
+  Undo adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project` <br>
+  `undo 2`<br>
+  Undo both commands.
 
-### 2.8. Clearing all entries : `clear`
+### 2.9. Redo a command : `redo`
 
-Clears all entries from the address book.<br>
-Format: `clear`
+Redo previously undone commands by the user.<br>
+Format: `redo [NUMBER]`
 
-### 2.9. Exiting the program : `exit`
+> Redo the last data-mutating command inputted since the undone point.<br>
+> If a number is entered, redo that ammount of previous commands instead.<br>
+> The number **must be a positive integer** 1, 2, 3, ...
+> The number must be less than or equal to the number of commands undone.
 
-Exits the program.<br>
-Format: `exit`
+Examples:
+
+* `add Test`<br>
+  `undo`<br>
+  `redo` <br>
+  Redo adding Test to the todo list.
+* `add Assignment` <br>
+  `add Project`<br>
+  `undo 2`<br>
+  `redo`<br>
+  Redo `add Assignment`.
 
 ### 2.10. Saving the data
 
-Address book data are saved in the hard disk automatically after any command that changes the data.<br>
+Todo list data are saved in the hard disk automatically after any command that changes the data.<br>
 There is no need to save manually.
 
-## 3. FAQ
+### 2.11. Viewing previous commands and accessing them : `history`
+
+Shows previous commands entered. <br>
+Format: `history`
 
-**Q**: How do I transfer my data to another Computer?<br>
-**A**: Install the app in the other computer and overwrite the empty data file it creates with
-       the file that contains the data of your previous Address Book folder.
+> * Previous commands are listed in order from latest command to earlier command.
+> * Alternatively, pressing on the <kbd>up</kbd> and <kbd>down</kbd> arrow keys on the keyboard will cycle through the commands previously entered.
+
+Examples:
+* `add Test` <br>
+  `history` <br>
+  Shows `add Test` in the list.
+* `add Test` <br>
+  You press on the <kbd>up</kbd> arrow key. <br>
+  Shows `add Test` in your input text field.
 
-## 4. Command Summary
+### 2.12. Add alias for any phrase: `alias`
 
-* **Add**  `add NAME p/PHONE_NUMBER e/EMAIL a/ADDRESS [t/TAG]...` <br>
-  e.g. `add James Ho p/22224444 e/jamesho@gmail.com a/123, Clementi Rd, 1234665 t/friend t/colleague`
+Adds an alias for a phrase. <br>
+Format: `alias ALIAS PHRASE`
+
+> * Once added, alias can be used instead of the phrase to perform operations.
+> * The phrase can be multiple words long
+
+Example:
+* `alias hs history` <br>
+  `hs` <br>
+  Shows `alias hs history` in the list.
+* `alias addTaskNamedTest add Test` <br>
```

Comment:
hmm kinda different from what we talked abt in #17 . should have quotes for the aliased command (so `alias addTaskNamedTest "add Test"`).